content/docs/context.md

```diff
@@ -4,220 +4,92 @@ title: Context
 permalink: docs/context.html
 ---
 
-> Note:
->
-> `React.PropTypes` has moved into a different package since React v15.5. Please use [the `prop-types` library instead](https://www.npmjs.com/package/prop-types) to define `contextTypes`.
->
->We provide [a codemod script](/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.proptypes) to automate the conversion.
-
-With React, it's easy to track the flow of data through your React components. When you look at a component, you can see which props are being passed, which makes your apps easy to reason about.
-
-In some cases, you want to pass data through the component tree without having to pass the props down manually at every level.
-You can do this directly in React with the powerful "context" API.
-
-> Note:
->
-> A [new, safe version of context](https://github.com/reactjs/rfcs/blob/master/text/0002-new-version-of-context.md) is under development for the upcoming 16.3 release.
+Context provides a way to pass data through the component tree without having to pass props down manually at every level.
 
+In a typical React application, data is passed top-down (parent to child) via props, but this can be cumbersome for certain types of props (e.g. locale preference, UI theme) that are required by many components within an application. Context provides a way to share values like this between components without having to explicitly pass a prop through every level of the tree.
 
-## Why Not To Use Context
+- [Motivation](#motivation)
+- [API](#api)
+  - [React.createContext](#reactcreatecontext)
+  - [Provider](#provider)
+  - [Consumer](#consumer)
+- [Examples](#examples)
+  - [Static Context](#static-context)
+  - [Dynamic Context](#dynamic-context)
+- [Legacy API](#legacy-api)
 
-The vast majority of applications do not need to use context.
 
-If you want your application to be stable, don't use context. It is an experimental API and it is likely to break in future releases of React.
+## Motivation
 
-If you aren't familiar with state management libraries like [Redux](https://github.com/reactjs/redux) or [MobX](https://github.com/mobxjs/mobx), don't use context. For many practical applications, these libraries and their React bindings are a good choice for managing state that is relevant to many components. It is far more likely that Redux is the right solution to your problem than that context is the right solution.
+Context is designed to relieve the pain of passing props down through a deeply nested component tree. For example, in the code below we manually thread through a color prop in order to style the Button and Message components:
```

Comment:
I anticipate that people new to React will assume from this paragraph that you need to use context as soon as you pass a prop down more than a few levels. I don't think we want to encourage that.

I think this should be more specific and say that context is for data that can be considered "global" for a tree of React components, and immediately give a few examples: currently authenticated user, current theme, current language.

Reply:
I think we should add a section "When to Use Context" that is very clear about intended use cases for this feature, as well as use cases where we *don't* recommend it (e.g. don't just use it to avoid passing props a few levels down).